src/protocols/ipfs-handler.js

```diff
@@ -94,11 +98,25 @@ export async function createHandler(ipfsOptions, session) {
               content: fs.createReadStream(filePath, { highWaterMark: 1024 * 1024 }),
             });
           }
+        } else if (data.type === "rawData" && data.bytes) {
+          // This contains the FormData field name/filename metadata
+          // Parse the rawData to extract the filename
+          const rawDataString = Buffer.from(data.bytes).toString('utf-8');
+          console.log("Raw data string:", rawDataString);
+          
+          // Extract filename from Content-Disposition header in the rawData
+          const filenameMatch = rawDataString.match(/filename="([^"]+)"/);
+          if (filenameMatch) {
+            currentFileName = filenameMatch[1];
+            console.log("Extracted filename from rawData:", currentFileName);
+          }
         } else if (data.type === "blob" && data.blobUUID) {
-          // Handle blob data
+          // Handle blob data from FormData - use the filename from previous rawData entry
           const blobData = await session.getBlobData(data.blobUUID);
-          const fileName = "index.html";
+          const fileName = currentFileName || "index.html";
+          console.log("Processing blob with filename:", fileName, "blobUUID:", data.blobUUID);
           entries.push({ path: fileName, content: blobData });
+          currentFileName = null; // Reset for next file
```

Comment:
Form-data uploads include a rawData part with the original filename, which we parse and store in currentFileName so the following blob chunk gets pushed with that relative path instead of defaulting to index.html. This ensures each IPFS entry keeps its original directory structure when published.